### PR TITLE
add more dsn params

### PR DESCRIPTION
**Pull request by hantmac**

### CODE REVIEW

File: restful.go [hunk 1]
```diff
@@ -165,7 +168,7 @@ func (c *APIClient) DoQuery(ctx context.Context, query string, args []driver.Val
 	request := QueryRequest{
 		SQL: q,
 		Pagination: Pagination{
-			WaitTime: 90,
+			WaitTime: c.WaitTimeSecs,
```
Comment (ZhiHanZ) [inline restful.go:171]: MaxRowsInBuffer      int64
	MaxRowsPerPage       int64
	need push to the pagination options similar to https://github.com/databendcloud/databend-jdbc/blob/main/databend-jdbc/src/main/java/com/databend/jdbc/DatabendConnection.java#L558

File: restful.go [hunk 2]
```diff
@@ -26,15 +26,15 @@ type APIClient struct {
...
+	User        string
+	Password    string
 	AccessToken string
+	Tenant      string
+	Warehouse   string
 
-	Tenant    string
-	Warehouse string
-
+	WaitTimeSecs         time.Duration
```
Comment (ZhiHanZ) [inline restful.go:35]: I think it is in integer unit as it counts based on seconds

**Review by everpcpc — APPROVED** [review]

File: restful.go (hunk 2 above)
Comment (hantmac) [reply restful.go:35]: In DSN, it used like this `"databend+https://username:password@tn3ftqihs.ch.aws-us-east-2.default.databend.com/test?timeout=1s&wait_time_secs=10s`

Comment (everpcpc) [reply restful.go:35]: It seems wired with both `_secs` in name and `s` in value

**Pull request merged**